actors/builtin/miner/expiration_queue.go

```diff
@@ -121,6 +121,31 @@ func (es *ExpirationSet) Count() (count uint64, err error) {
 	return onTime + early, nil
 }
 
+// validates a set of assertions that must hold for expiration sets
+func (es *ExpirationSet) ValidateState() error {
+	if es.OnTimePledge.LessThan(big.Zero()) {
+		return xerrors.Errorf("ESet left with negative pledge: %+v", es)
+	}
+
+	if es.ActivePower.Raw.LessThan(big.Zero()) {
+		return xerrors.Errorf("ESet left with negative raw active power: %+v", es)
+	}
+
+	if es.ActivePower.QA.LessThan(big.Zero()) {
+		return xerrors.Errorf("ESet left with negative qa active power: %+v", es)
+	}
+
+	if es.FaultyPower.Raw.LessThan(big.Zero()) {
+		return xerrors.Errorf("ESet left with negative raw faulty power: %+v", es)
+	}
+
+	if es.FaultyPower.QA.LessThan(big.Zero()) {
+		return xerrors.Errorf("ESet left with negative qa active power: %+v", es)
+	}
```

Comment:
Whoops, typo in the error string. My bad there! Should be "negative qa faulty power"